XenModel/Network/Cache.cs

```diff
@@ -564,22 +565,23 @@ public void CheckDockerContainersBatchChange()
         }
 
         private bool dockerContainersChanged = false;
-        public void UpdateDockerContainersForVM(IEnumerable<DockerContainer> containers, VM vm)
+        public void UpdateDockerContainersForVM(IList<DockerContainer> containers, VM vm)
         {
-            dockerContainersChanged = true;
-         
+            Trace.Assert(vm != null);
+
             //updating existing, adding new containers
+            dockerContainersChanged = dockerContainersChanged || containers.Count > 0;

```

Comment:
It definitely would, but it is not there in C#. 
I was not sure, so did a bit of research and it seems the bitwise operator |= could be used instead, although that one is not short circuiting as || is, so I would say I would leave it as is.